lib/ex_gram/updates/webhook.ex

```diff
@@ -0,0 +1,131 @@
+defmodule ExGram.Updates.Webhook do
+  @moduledoc """
+  Updates implementation that uses webhook method
+  """
+
+  use GenServer
+  require Logger
+
+  @posible_updates Map.keys(%ExGram.Model.Update{})
+                   |> List.delete(:__struct__)
+                   |> List.delete(:update_id)
+
+  def update(update, token_hash) do
+    Module.concat(__MODULE__, token_hash)
+    |> GenServer.cast({:update, update})
+  end
+
+  def start_link({:bot, pid, :token, token}) do
+    name = Module.concat(__MODULE__, token_hash(token))
+
+    GenServer.start_link(__MODULE__, {:ok, pid, token}, name: name)
+  end
+
+  def init({:ok, pid, token}) do
+    set_webhook(token)
+
+    {:ok, {pid, token}}
+  end
+
+  def handle_cast({:update, update}, {pid, token}) do
+    GenServer.call(pid, {:update, update})
+
+    {:noreply, {pid, token}}
+  end
+
+  def handle_info(unknown_message, state) do
+    Logger.debug("Webhook updates received an unknown message #{inspect(unknown_message)}")
+
+    {:noreply, state}
+  end
+
+  defp token_hash(token) do
+    :crypto.hash(:sha, token)
+    |> Base.url_encode64(padding: true)
+  end
+
+  defp set_webhook(token) do
+    config = ExGram.Config.get(:ex_gram, :webhook)
+    params = webhook_params(config)
+
+    case config[:url] do
+      webhook_url when is_binary(webhook_url) ->
+        webhook_url =
+          webhook_url
+          |> Plug.Router.Utils.split()
+
+        case ExGram.set_webhook(
+               "https://#{webhook_url}/telegram/#{token_hash(token)}",
+               [{:token, token} | params]
+             ) do
+          {:ok, _} -> nil
+          {:error, error} -> Logger.error("Could not set the webhook: #{inspect(error)}")
+        end
+
+      nil ->
+        Logger.warning(
+          "The webhook_url is not set in the configuration. Please manually set the webhook using this method: https://core.telegram.org/bots/api#setwebhook"
+        )
+    end
+  end
+
+  defp webhook_params(_, params \\ [])
+  defp webhook_params([], params), do: params
+
+  defp webhook_params([{:certificate, path} | tl], params) do
+    case File.read(path) do
+      {:ok, _} ->
+        webhook_params(tl, [{:certificate, {:file, path}} | params])
+
+      {:error, reason} ->
+        Logger.error("Could not read the certificate file from #{path}: #{inspect(reason)}")
+
+        webhook_params(tl, params)
+    end
+  end
+
+  defp webhook_params([{:url, _} | tl], params), do: webhook_params(tl, params)
+
+  defp webhook_params([{:max_connections, max_connections} | tl], params)
+       when is_integer(max_connections),
+       do: webhook_params(tl, [{:max_connections, max_connections} | params])
```

Comment:
When the first `def` line is too long, I like to use full `do` block instead of `do:`

```suggestion
  defp webhook_params([{:max_connections, max_connections} | tl], params)
       when is_integer(max_connections) do
       webhook_params(tl, [{:max_connections, max_connections} | params])
  end
```